Assets/Countly/Plugins/CountlySDK/Services/ViewCountlyService.cs

```diff
@@ -8,24 +8,723 @@
+            currentViewData.ViewStartTimeSeconds = _utils.CurrentTimestampSeconds();
+            currentViewData.IsAutoStoppedView = viewShouldBeAutomaticallyStopped;
+
+            viewDataMap.Add(currentViewData.ViewID, currentViewData);
+            previousViewID = currentViewID;
+            currentViewID = currentViewData.ViewID;
+
+            Dictionary<string, object> accumulatedEventSegm = new Dictionary<string, object>(automaticViewSegmentation);
+
+            if (customViewSegmentation != null) {
+                foreach (KeyValuePair<string, object> kvp in customViewSegmentation) {
+                    accumulatedEventSegm.Add(kvp.Key, kvp.Value);
+                }
+            }
+
+            Dictionary<string, object> viewSegmentation = CreateViewEventSegmentation(currentViewData, _isFirstView, true, accumulatedEventSegm);
+
+            if (_isFirstView) {
+                Log.Debug("[ViewCountlyService] Recording view as the first one in the session. [" + viewName + "]");
+                _isFirstView = false;
+            }
+            _ = _eventService.RecordEventInternal(viewEventKey, viewSegmentation, 1, 0, null, currentViewData.ViewID);
+
+            return currentViewData.ViewID;
+        }
+
+        /// <summary>
+        /// Stops a view with the given name if it was open.
+        /// </summary>
+        /// <param name="viewName">name of the view</param>
+        /// <param name="customViewSegmentation"></param>
+        private void StopViewWithNameInternal(string viewName, Dictionary<string, object> customViewSegmentation)
+        {
+            if (_utils.IsNullEmptyOrWhitespace(viewName)) {
+                Log.Warning("[ViewCountlyService] StopViewWithNameInternal, Trying to record view with null or empty view name, ignoring request");
+                return;
+            }
+
+            string viewID = null;
+
+            foreach (KeyValuePair<string, ViewData> entry in viewDataMap) {
+                string key = entry.Key;
+                ViewData vd = entry.Value;
+
+                if (vd != null && viewName.Equals(vd.ViewName)) {
+                    viewID = key;
+                }
+            }
+
+            if (viewID == null) {
+                Log.Warning("[ViewCountlyService] StopViewWithNameInternal, No view entry found with the provided name :[" + viewName + "]");
+                return;
+            }
+
+            StopViewWithIDInternal(viewID, customViewSegmentation);
+        }
+
+        /// <summary>
+        /// Closes given views or all views.
+        /// </summary>
+        /// <param name="closeAllViews"></param>
+        /// <param name="customViewSegmentation"></param>
+        private void AutoCloseRequiredViews(bool closeAllViews, Dictionary<string, object> customViewSegmentation)
+        {
+            Log.Debug("[ViewCountlyService] AutoCloseRequiredViews");
+            List<string> viewsToRemove = new List<string>(1);
```

Comment:
this 1 has same logic like Java or different?

Reply:
seems like same/similar